core/federated/RTI/rti_remote.c

```diff
@@ -1287,7 +1309,36 @@ static int32_t receive_and_check_fed_id_message(int *socket_id, struct sockaddr_
         // Terminate the string with a null.
         federation_id_received[federation_id_length] = 0;
 
-        LF_PRINT_DEBUG("RTI received federation ID: %s.", federation_id_received);
+        if (buffer[0] == MSG_TYPE_FED_IDS) {
+            // Legacy federate, pre-versioning. Set the protocol version to 0 unless it
+            // has been previously set.
+            if (rti_remote->protocol_version > 0) {
+                // The protocol version has already been set to an incompatible version.
+                LF_PRINT_DEBUG("Federate using protocol version 0 rejected.");
+                send_reject(socket_id, LF_FED_VERSION_DOES_NOT_MATCH);
+                return -1;
+            }
+            rti_remote->protocol_version = 0;
+            LF_PRINT_DEBUG("RTI: Legacy protocol version 0 will be used.");
+        } else {
+            // Extract the proposed protocol version.
+            int32_t proposed_version = 
+                    ((int32_t)buffer[1]) << 16 
+                    | ((int32_t)buffer[2]) << 8 
+                    | ((int32_t)buffer[3]);
+            if (rti_remote->protocol_version > 0 && proposed_version != rti_remote->protocol_version) {
```

Comment:
It's good that we use semantic versions, but then we shouldn't require exact matches. According to semver, any minor or patch increments should not be include breaking changes. We haven't done a good job with this in the `lingua-franca` repo, but we should really try to adhere to the semver rules here...